File: Editorial/66_Plus_One/JavaScript/1_Schoolbook_Addition_with_Carry.js
var plusOne = function (digits) {
    let n = digits.length;
    //move along the input array starting from the end
    for (let i = n - 1; i >= 0; --i) {
        //set all the nines at the end of array to zeros
        if (digits[i] == 9) {
            digits[i] = 0;
        }
        //here we have the rightmost not-nine
        else {
            //Increase this rightmost not-nine by 1
            digits[i]++;
            //and the job is done
            return digits;
        }
    }
    //We're here because all the digits are nines
    digits.unshift(1);
    return digits;
};

File: Editorial/0066_Plus_One/Go/1_Schoolbook_Addition_with_Carry.go
func plusOne(digits []int) []int {
    n := len(digits)
    // move along the input array starting from the end
    for i := n - 1; i >= 0; i-- {
        // set all the nines at the end of array to zeros
        if digits[i] == 9 {
            digits[i] = 0
        } else {
            // here we have the rightmost not-nine
            // increase this rightmost not-nine by 1
            digits[i]++
            // and the job is done
            return digits
        }
    }
    // We're here because all the digits are nines
    digits = append([]int{1}, digits...)
    return digits
}
